src/ResourceManager/DevTestLabs/Commands.DevTestLabs/Commands/Set/SetAzureRmDtlAllowedVMSizesPolicy.cs

```diff
@@ -0,0 +1,122 @@
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.DevTestLabs.Models;
+using Microsoft.Azure.Commands.DevTestLabs.Properties;
+using Microsoft.Azure.Management.DevTestLabs;
+using Microsoft.Azure.Management.DevTestLabs.Models;
+using Microsoft.Rest.Azure;
+using Newtonsoft.Json;
+using System;
+using System.Globalization;
+using System.Linq;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.DevTestLabs
+{
+    [Cmdlet(VerbsCommon.Set, "AzureRmDtlAllowedVMSizesPolicy", HelpUri = Constants.DevTestLabsHelpUri, DefaultParameterSetName = ParameterSetEnable, SupportsShouldProcess = true)]
+    [OutputType(typeof(PSPolicy))]
+    public class SetAzureRmDtlAllowedVMSizesPolicy : DtlPolicyCmdletBase
+    {
+        #region Input Parameter Definitions
+
+        /// <summary>
+        /// Maximum number of virtual machines.
+        /// </summary>
+        [Parameter(Mandatory = false,
+            Position = 4,
+            HelpMessage = "Maximum number of virtual machines a user is allowed to own in the lab.")]
+        public string[] VmSizes { get; set; }
+
+        protected override string PolicyName
+        {
+            get
+            {
+                return WellKnownPolicyNames.AllowedVmSizesInLab;
+            }
+        }
+
+        #endregion Input Parameter Definitions
+
+        public override void ExecuteCmdlet()
+        {
+            Policy inputPolicy = null;
+
+            try
+            {
+                inputPolicy = DataServiceClient.Policy.GetResource(
+                                ResourceGroupName,
+                                LabName,
+                                Constants.Default,
+                                PolicyName);
+            }
+            catch (CloudException ex)
+            {
+                if (ex.Response.StatusCode != System.Net.HttpStatusCode.NotFound
+                    || VmSizes == null)
+                {
+                    throw;
+                }
+            }
+
+            if (inputPolicy == null)
+            {
+                inputPolicy = new Policy
+                {
+                    FactName = PolicyFactName.LabVmSize,
+                    Threshold = JsonConvert.SerializeObject(VmSizes),
+                    EvaluatorType = PolicyEvaluatorType.AllowedValuesPolicy,
+                    Status = Disable ? PolicyStatus.Disabled : PolicyStatus.Enabled
+                };
+            }
+            else
+            {
+                // Do nothing if user cancelled the operation
+                var actionDescription = string.Format(CultureInfo.CurrentCulture, Resources.SavePolicyDescription, PolicyName, LabName);
+                var actionWarning = string.Format(CultureInfo.CurrentCulture, Resources.SavePolicyWarning, PolicyName);
+                if (!ShouldProcess(
```

Comment:
This is in the wrong place, it needs to protect the actual CreateOrUpdate call, regardless of whether input policy is null